ui/pages/snaps/snap-view/snap-view.js

```diff
@@ -0,0 +1,340 @@
+  showKeyringSnapRemovalModal,
+  getSnapAccountsById,
+  ///: END:ONLY_INCLUDE_IN
+} from '../../../store/actions';
+import {
+  getSnaps,
+  getSubjectsWithSnapPermission,
+  getPermissions,
+  getPermissionSubjects,
+  getTargetSubjectMetadata,
+  ///: BEGIN:ONLY_INCLUDE_IN(keyring-snaps)
+  getMemoizedMetaMaskIdentities,
+  ///: END:ONLY_INCLUDE_IN
+} from '../../../selectors';
+import { getSnapName } from '../../../helpers/utils/util';
+import {
+  Box,
+  Button,
+  ButtonIcon,
+  ButtonSize,
+  ButtonVariant,
+  Text,
+} from '../../../components/component-library';
+import SnapPermissionsList from '../../../components/app/snaps/snap-permissions-list';
+import { SnapDelineator } from '../../../components/app/snaps/snap-delineator';
+import { DelineatorType } from '../../../helpers/constants/snaps';
+import {
+  Content,
+  Header,
+  Page,
+} from '../../../components/multichain/pages/page';
+///: BEGIN:ONLY_INCLUDE_IN(keyring-snaps)
+import { KeyringSnapRemovalResultStatus } from './constants';
+///: END:ONLY_INCLUDE_IN
+
+function SnapView() {
+  const t = useI18nContext();
+  const history = useHistory();
+  const location = useLocation();
+  const descriptionRef = useRef(null);
+  const { pathname } = location;
+  // The snap ID is in URI-encoded form in the last path segment of the URL.
+  const decodedSnapId = decodeURIComponent(pathname.match(/[^/]+$/u)[0]);
+  const snaps = useSelector(getSnaps);
+  const snap = Object.entries(snaps)
+    .map(([_, snapState]) => snapState)
+    .find((snapState) => snapState.id === decodedSnapId);
+
+  const [isShowingRemoveWarning, setIsShowingRemoveWarning] = useState(false);
+  const [isDescriptionOpen, setIsDescriptionOpen] = useState(false);
+  const [isOverflowing, setIsOverflowing] = useState(false);
+  // eslint-disable-next-line no-unused-vars -- Main build does not use setIsRemovingKeyringSnap
+  const [isRemovingKeyringSnap, setIsRemovingKeyringSnap] = useState(false);
+
+  // eslint-disable-next-line no-unused-vars -- Main build does not use setKeyringAccounts
+  const [keyringAccounts, setKeyringAccounts] = useState([]);
+  ///: BEGIN:ONLY_INCLUDE_IN(keyring-snaps)
+  const identities = useSelector(getMemoizedMetaMaskIdentities);
+  ///: END:ONLY_INCLUDE_IN
+
+  useEffect(() => {
+    if (!snap) {
+      history.push(SNAPS_ROUTE);
+    }
+  }, [history, snap]);
+
+  useEffect(() => {
+    setIsOverflowing(
+      descriptionRef.current &&
+        descriptionRef.current.offsetHeight <
+          descriptionRef.current.scrollHeight,
+    );
+  }, [descriptionRef]);
+
+  const connectedSubjects = useSelector((state) =>
+    getSubjectsWithSnapPermission(state, snap?.id),
+  );
+  const permissions = useSelector(
+    (state) => snap && getPermissions(state, snap.id),
+  );
+  const subjects = useSelector((state) => getPermissionSubjects(state));
+  const targetSubjectMetadata = useSelector((state) =>
+    getTargetSubjectMetadata(state, snap?.id),
+  );
+
+  let isKeyringSnap = false;
```

Comment:
Just to be safe, let's have someone from the accounts team review this PR too

Reply:
Tested locally and everything is ok.